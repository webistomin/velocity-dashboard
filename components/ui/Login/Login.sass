+b('login')
  display: flex
  flex-direction: column
  flex-grow: 1

  +e('container')
    display: inherit
    flex-direction: inherit
    flex-grow: inherit
    justify-content: center
    padding-bottom: rem(52px)

  +e('title')
    text-align: center

    +media($screen-sm)
      text-align: left

  +e('desc')
    text-align: center

    +media($screen-sm)
      text-align: left

  +e('heading')
    margin-bottom: rem(48px)

    +media($screen-sm)
      margin-bottom: rem(30px)

    +media($screen-lg)
      margin-bottom: rem(45px)

  +e('form-inputs')
    margin-bottom: rem(25px)

  +e('form-group')
    &:not(:last-of-type)
      margin-bottom: rem(20px)

  +e('submit')
    margin-bottom: rem(14px)

  +e('text')
    text-align: center

  +e('changer')
    margin-left: 0.5ch

  +e('col')
    +m('left')
      position: relative

      +media($screen-sm)
        width: calc(66vw - 14%)
        display: flex
        flex-direction: column

        +media($screen-lg)
          width: 41%

    +m('right')
      display: none

      +media($screen-sm)
        display: block
        position: fixed
        top: 0
        right: 0
        width: 32vw
        height: 100vh

      +media($screen-lg)
        width: 50%

  +e('holder')
    backface-visibility: hidden
    will-change: opacity, position

    +media($screen-sm)
      width: rem(320px)
      margin: auto

  +e('list')
    display: grid
    grid-template-columns: repeat(auto-fill, minmax(46%, 1fr))
    grid-gap: 20px
    margin-bottom: rem(20px)

  +e('option-name')
    font: var(--text-button)
    padding-bottom: rem(6px)

  +e('option-desc')
    margin-bottom: auto

  +e('option-content')
    min-height: rem(196px)
    box-sizing: border-box
    padding: rem(20px)
    display: flex
    height: 100%
    flex-direction: column
    justify-content: center
    align-items: center
    text-align: center

  +e('option-icon')
    margin-bottom: rem(15px)

  +e('inner')
    +media($screen-sm)
      display: flex
      height: 100%
      flex-grow: 1

  +e('video')
    width: 100%
    height: 100%
    display: block
    object-fit: cover

  +e('video-block')
    width: 100%
    height: 100%
    position: relative
    background-image: url('/img/login-poster.jpg')
    background-size: cover
    background-position: center
    background-repeat: no-repeat

    &::before
      content: ''
      z-index: 2
      position: absolute
      top: 0
      left: 0
      width: 100%
      height: 100%
      background-color: var(--color-primary)
      opacity: 0.2

