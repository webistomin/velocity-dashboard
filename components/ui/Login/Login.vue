<template lang="pug">
  section.login
    .login__container.container
      .login__inner
        .login__col.login__col_left
          BaseTitle(:level='3').login__title Sign in to Velocity
          p.login__desc.paragraph.paragraph_color_darkgray Please enter your credentials to proceed.
          form.login__form
            .login__form-inputs
              BaseFormGroup(
                :type='"email"'
                :label='"Email address"'
                :id='"sign-in-email"'
                :placeholder='"Enter your email"'
                :validator='$v.signInForm.email'
                @input='signInForm.email = $event'
                :value='signInForm.email'
                @blur='$v.signInForm.email.$touch()'
              ).login__form-group
              BaseFormGroup(
                :type='"password"'
                :label='"Password"'
                :id='"sign-in-password"'
                :placeholder='"Enter your password"'
                :validator='$v.signInForm.password'
                @input='signInForm.password = $event'
                :value='signInForm.password'
                @blur='$v.signInForm.password.$touch()'
              ).login__form-group
            BaseButton(
              :type='"submit"'
            ) Sign in
        .login__col.login__col_right
</template>

<script lang="ts">
  import Vue from 'vue';
  import Component from 'vue-class-component';
  // @ts-ignore
  import { email, required } from 'vuelidate/lib/validators';
  import BaseTitle from 'components/base/BaseTitle';
  import BaseFormGroup from 'components/base/BaseFormGroup';
  import BaseButton from 'components/base/BaseButton';

  @Component({
    name: 'Login',
    components: { BaseTitle, BaseFormGroup, BaseButton },
    validations: {
      signInForm: {
        email: {
          required,
          email,
        },
        password: {
          required,
        },
      },
    },
  })
  export default class Login extends Vue {
    public data() {
      return {
        signInForm: {
          email: '',
          password: '',
        },
      };
    }
  }
</script>

<style lang="sass">
  @import "Login"
</style>
