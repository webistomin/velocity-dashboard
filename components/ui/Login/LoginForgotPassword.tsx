import { VueComponent } from 'types/vue-components';
import { Component, Emit } from 'nuxt-property-decorator';
import { VNode } from 'vue';
import { email, required } from 'vuelidate/lib/validators';
import { IUserInterface } from 'common/types/user/user-schema';
import { serverUrls } from 'common/urls/serverUrls';
import { IAuthResetResponseBody } from 'common/types/auth/reset';

import BaseTitle from 'components/base/BaseTitle';
import BaseFormGroup from 'components/base/BaseFormGroup';
import BaseButton from 'components/base/BaseButton';
import BaseLink from 'components/base/BaseLink';
import { FormTypes, ILoginProps } from './Login';

export interface ILoginForgotForm {
  email: IUserInterface['email'];
}

@Component({
  name: 'LoginForgot',
  validations: {
    forgotForm: {
      email: {
        required,
        email,
      },
    },
  },
})
export default class LoginForgot extends VueComponent<ILoginProps> {
  public forgotForm: ILoginForgotForm = {
    email: '',
  };

  public isLoading: boolean = false;

  @Emit('setFormName')
  public setFormType(type: FormTypes): FormTypes {
    return type;
  }

  public async onSubmit(event: Event) {
    event.preventDefault();

    const validator = this.$v;
    validator.$touch();

    if (!validator.$anyError) {
      this.isLoading = true;

      try {
        const data = this.forgotForm;
        await this.$axios.$post(serverUrls.auth.reset, data).then((response: IAuthResetResponseBody) => {
          this.isLoading = false;
          if (response.success) {
            this.$notify({
              group: 'auth',
              type: 'success',
              title: 'Success',
              text: 'Link was sent to your email',
              duration: 3000,
            });
          }
        });
      } catch (e) {
        this.isLoading = false;
        this.$notify({
          group: 'auth',
          type: 'error',
          title: 'Authentication error',
          text: e?.response?.data?.message,
          duration: 3000,
        });
      }
    }
  }

  public render(): VNode {
    return (
      <div class='login__holder'>
        <div class='login__heading'>
          <BaseTitle class='login__title' level={3}>
            Recover password
          </BaseTitle>
          <p class='login__desc paragraph paragraph_color_darkgray'>Don't worry, happens to the best of us.</p>
        </div>
        <form class='login__form' onSubmit={this.onSubmit}>
          <div class='login__form-inputs'>
            <BaseFormGroup
              class='login__form-group'
              type='email'
              label='Email address'
              id='forgot-email'
              placeholder='Enter your email'
              validator={this.$v.forgotForm.email}
              onInput={($event: string) => (this.forgotForm.email = $event)}
              value={this.forgotForm.email}
              onBlur={this.$v.forgotForm.email?.$touch}
            />
          </div>
          <BaseButton class='login__submit' type='submit' isLoading={this.isLoading}>
            Reset my password
          </BaseButton>
          <p class='login__text paragraph'>
            Back to
            <BaseLink class='login__changer' isButton={true} onClick={() => this.setFormType(FormTypes.SIGN_IN)}>
              Sign in
            </BaseLink>
          </p>
        </form>
      </div>
    );
  }
}
