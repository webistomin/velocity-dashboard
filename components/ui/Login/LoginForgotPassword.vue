<template lang="pug">
  .login__holder
    .login__heading
      BaseTitle(:level='3').login__title Recover password
      p.login__desc.paragraph.paragraph_color_darkgray Don't worry, happens to the best of us.
    form.login__form
      .login__form-inputs
        BaseFormGroup(
          :type='"email"'
          :label='"Email address"'
          :id='"forgot-email"'
          :placeholder='"Enter your email"'
          :validator='$v.forgotForm.email'
          @input='forgotForm.email = $event'
          :value='forgotForm.email'
          @blur='$v.forgotForm.email.$touch()'
        ).login__form-group
      BaseButton(
        :type='"submit"'
      ).login__submit Reset my password
      p.login__text.paragraph Back to
        BaseLink(
          :isButton='true'
          @click='setFormType(FormTypes.SIGN_IN)'
          ).login__changer Sign in
</template>

<script lang="ts">
  import Vue from 'vue';
  import Component from 'vue-class-component';
  // @ts-ignore
  import { email, required } from 'vuelidate/lib/validators';
  import BaseFormGroup from 'components/base/BaseFormGroup';
  import BaseButton from 'components/base/BaseButton';
  import BaseLink from 'components/base/BaseLink';
  import { FormTypes } from 'components/ui/Login/Login.vue';

  @Component({
    name: 'LoginForgot',
    components: { BaseFormGroup, BaseButton, BaseLink },
    validations: {
      forgotForm: {
        email: {
          required,
          email,
        },
      },
    },
  })
  export default class LoginForgot extends Vue {
    FormTypes: any = FormTypes;

    public data() {
      return {
        forgotForm: {
          email: '',
        },
      };
    }

    public setFormType(type: FormTypes): void {
      this.$emit('setFormName', type);
    }
  }
</script>
