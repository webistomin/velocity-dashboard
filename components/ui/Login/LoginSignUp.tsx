import { VueComponent } from 'types/vue-components';
import { Component, Emit } from 'nuxt-property-decorator';
import { VNode } from 'vue';
import { email, required } from 'vuelidate/lib/validators';
// eslint-disable-next-line @typescript-eslint/no-unused-vars,no-unused-vars

import BaseTitle from 'components/base/BaseTitle';
import BaseFormGroup from 'components/base/BaseFormGroup';
import BaseButton from 'components/base/BaseButton';
import BaseLink from 'components/base/BaseLink';
import BaseOption from 'components/base/BaseOption';
import BaseIcon from 'components/base/BaseIcon';

import { FormTypes, ILoginProps } from './Login';

@Component({
  name: 'LoginSignUp',
  validations: {
    signUpForm: {
      name: {
        required,
      },
      email: {
        required,
        email,
      },
      password: {
        required,
      },
    },
  },
})
export default class LoginSignUp extends VueComponent<ILoginProps> {
  signUpForm = {
    name: '',
    email: '',
    password: '',
    role: 'Administrator',
  };

  roles = [
    {
      name: 'Administrator',
      desc: 'Full access to all settings',
      icon: 'icon-lightning',
    },
    {
      name: 'Operator',
      desc: 'Service desk and chat permissions',
      icon: 'icon-bubble',
    },
  ];

  @Emit('setFormName')
  public setFormType(type: FormTypes): FormTypes {
    return type;
  }

  updateRoleValue(event: Event) {
    const target = event.target as HTMLInputElement;
    this.signUpForm.role = target.value;
  }

  render(): VNode {
    return (
      <div class='login__holder'>
        <div class='login__heading'>
          {/*
          // @ts-ignore */}
          <BaseTitle class='login__title' level={3}>
            Get started for free
          </BaseTitle>
          <p class='login__desc paragraph paragraph_color_darkgray'>Free forever. No credit card needed.</p>
        </div>
        <form class='login__form'>
          <ul class='login__list list'>
            {this.roles.map((role) => {
              return (
                <li class='login__item list-item' key={role.name}>
                  <BaseOption
                    class='login__option'
                    type='radio'
                    value={role.name}
                    name='login-role'
                    id={`login-role-${role.name}`}
                    onInput={(event: Event) => this.updateRoleValue(event)}
                    checked={role.name === this.signUpForm.role}>
                    <span class='login__option-content'>
                      {/*
                      // @ts-ignore */}
                      <BaseIcon size='s' name={role.icon} color='default' class='login__option-icon' />
                      <strong class='login__option-name'>{role.name}</strong>
                      <span class='login__option-desc paragraph paragraph_color_darkgray'>{role.desc}</span>
                    </span>
                  </BaseOption>
                </li>
              );
            })}
          </ul>
          <div class='login__form-inputs'>
            <BaseFormGroup
              class='login__form-group'
              type='text'
              label='Full name'
              id='sign-up-name'
              placeholder='Enter your full name'
              validator={this.$v.signUpForm.name}
              onInput={($event: string) => (this.signUpForm.name = $event)}
              value={this.signUpForm.name}
              onBlur={this.$v.signUpForm.name?.$touch}
            />
            <BaseFormGroup
              class='login__form-group'
              type='email'
              label='Email address'
              id='sign-up-email'
              placeholder='Enter your email'
              validator={this.$v.signUpForm.email}
              onInput={($event: string) => (this.signUpForm.email = $event)}
              value={this.signUpForm.email}
              onBlur={this.$v.signUpForm.email?.$touch}
            />
            <BaseFormGroup
              class='login__form-group'
              type='password'
              label='Password'
              id='sign-up-password'
              placeholder='Enter your password'
              validator={this.$v.signUpForm.password}
              onInput={($event: string) => (this.signUpForm.password = $event)}
              value={this.signUpForm.password}
              onBlur={this.$v.signUpForm.password?.$touch}
            />
          </div>
          <BaseButton class='login__submit' type='button'>
            Create Account
          </BaseButton>
          <p class='login__text paragraph'>
            Already have an account?
            <BaseLink class='login__changer' isButton={true} onClick={() => this.setFormType(FormTypes.SIGN_IN)}>
              Sign in
            </BaseLink>
          </p>
        </form>
      </div>
    );
  }
}
