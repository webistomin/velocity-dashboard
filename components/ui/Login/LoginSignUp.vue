<template lang="pug">
  .login__holder
    .login__heading
      BaseTitle(:level='3').login__title Get started for free
      p.login__desc.paragraph.paragraph_color_darkgray Free forever. No credit card needed.
    form.login__form
      .login__form-inputs
        BaseFormGroup(
          :type='"text"'
          :label='"Full name"'
          :id='"sign-up-name"'
          :placeholder='"Enter your full name"'
          :validator='$v.signUpForm.name'
          @input='signUpForm.name = $event'
          :value='signUpForm.name'
          @blur='$v.signUpForm.name.$touch()'
        ).login__form-group
        BaseFormGroup(
          :type='"email"'
          :label='"Email address"'
          :id='"sign-up-email"'
          :placeholder='"Enter your email"'
          :validator='$v.signUpForm.email'
          @input='signUpForm.email = $event'
          :value='signUpForm.email'
          @blur='$v.signUpForm.email.$touch()'
        ).login__form-group
        BaseFormGroup(
          :type='"password"'
          :label='"Password"'
          :id='"sign-up-password"'
          :placeholder='"Enter your password"'
          :validator='$v.signUpForm.password'
          @input='signUpForm.password = $event'
          :value='signUpForm.password'
          @blur='$v.signUpForm.password.$touch()'
        ).login__form-group
      BaseButton(
        :type='"submit"'
      ).login__submit Create Account
      p.login__text.paragraph Already have an account?
        BaseLink(
          :isButton='true'
          @click='setFormType(FormTypes.SIGN_IN)'
          ).login__changer Sign in
</template>

<script lang="ts">
  import Vue from 'vue';
  import Component from 'vue-class-component';
  // @ts-ignore
  import { email, required } from 'vuelidate/lib/validators';
  import BaseFormGroup from 'components/base/BaseFormGroup';
  import BaseButton from 'components/base/BaseButton';
  import BaseLink from 'components/base/BaseLink';
  import { FormTypes } from './Login.vue';

  @Component({
    name: 'LoginSignUp',
    components: { BaseFormGroup, BaseButton, BaseLink },
    validations: {
      signUpForm: {
        name: {
          required,
        },
        email: {
          required,
          email,
        },
        password: {
          required,
        },
      },
    },
  })
  export default class LoginSignUp extends Vue {
    FormTypes: any = FormTypes;

    public data() {
      return {
        signUpForm: {
          name: '',
          email: '',
          password: '',
        },
      };
    }

    public setFormType(type: FormTypes): void {
      this.$emit('setFormName', type);
    }
  }
</script>
