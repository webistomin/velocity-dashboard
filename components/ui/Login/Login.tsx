import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import { VueComponent } from 'types/vue-components';

import LoginSignIn from './LoginSignIn';
import LoginSignUp from './LoginSignUp';
import LoginForgotPassword from './LoginForgotPassword';
import LoginVideo from './LoginVideo';
import LoginHeader from './LoginHeader';

import './Login.sass';

export enum FormTypes {
  SIGN_IN = 'sign-in',
  SIGN_UP = 'sign-up',
  FORGOT_PASSWORD = 'forgot-password',
}

export interface ILoginProps {
  onSetFormName: (type: FormTypes) => void;
}

@Component({
  name: 'Login',
})
export default class Login extends VueComponent {
  public currentForm = FormTypes.SIGN_IN;

  public setFormType(formName: FormTypes): void {
    this.currentForm = formName;
  }

  public render(): VNode {
    return (
      <section class='login'>
        <h2 class='visually-hidden'>Authorization</h2>
        <div class='login__container container'>
          <div class='login__inner'>
            <div class='login__col login__col_left'>
              <LoginHeader />
              <transition name='login-fade' mode='out-in'>
                {this.currentForm === FormTypes.SIGN_IN && (
                  <LoginSignIn onSetFormName={(formName) => this.setFormType(formName)} />
                )}
                {this.currentForm === FormTypes.SIGN_UP && (
                  <LoginSignUp onSetFormName={(formName) => this.setFormType(formName)} />
                )}
                {this.currentForm === FormTypes.FORGOT_PASSWORD && (
                  <LoginForgotPassword onSetFormName={(formName) => this.setFormType(formName)} />
                )}
              </transition>
            </div>
            <div class='login__col login__col_right'>
              <LoginVideo />
            </div>
          </div>
        </div>
      </section>
    );
  }
}
