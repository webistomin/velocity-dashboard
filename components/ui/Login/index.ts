import Login from './Login.vue';
import LoginSignIn from './LoginSignIn.vue';
import LoginSignUp from './LoginSignUp.vue';
import LoginForgotPassword from './LoginForgotPassword.vue';

export { Login, LoginSignIn, LoginSignUp, LoginForgotPassword };

export default Login;
