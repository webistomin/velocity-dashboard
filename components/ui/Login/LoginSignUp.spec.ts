import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuelidate from 'vuelidate';
import { FormTypes } from 'components/ui/Login/Login';
import LoginSignUp from './LoginSignUp';

describe('LoginSignUp', () => {
  const localVue = createLocalVue();
  localVue.use(Vuelidate);
  const factory = (propsData = {}, options = {}) => {
    return shallowMount(LoginSignUp, {
      localVue,
      propsData: {
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Emit "setFormName" event', () => {
    const wrapper = factory();
    const result = wrapper.vm.setFormType(FormTypes.SIGN_IN);
    expect(wrapper.emitted().setFormName);
    expect(result).toEqual(FormTypes.SIGN_IN);
  });
});
