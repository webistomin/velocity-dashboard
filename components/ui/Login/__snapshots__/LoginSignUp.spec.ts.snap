// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginSignUp Matches snapshot 1`] = `
"<div class=\\"login__holder\\">
  <div class=\\"login__heading\\">
    <vuecomponent-stub level=\\"3\\" class=\\"login__title\\">Get started for free</vuecomponent-stub>
    <p class=\\"login__desc paragraph paragraph_color_darkgray\\">Free forever. No credit card needed.</p>
  </div>
  <form class=\\"login__form\\">
    <ul class=\\"login__list list\\">
      <li class=\\"login__item list-item\\">
        <baseoption-stub type=\\"radio\\" value=\\"admin\\" name=\\"login-role\\" id=\\"login-role-admin\\" class=\\"login__option\\"><span class=\\"login__option-content\\"><vuecomponent-stub size=\\"s\\" name=\\"icon-lightning\\" color=\\"default\\" class=\\"login__option-icon\\"></vuecomponent-stub><strong class=\\"login__option-name\\">Administrator</strong><span class=\\"login__option-desc paragraph paragraph_color_darkgray\\">Full access to all settings</span></span></baseoption-stub>
      </li>
      <li class=\\"login__item list-item\\">
        <baseoption-stub type=\\"radio\\" value=\\"operator\\" name=\\"login-role\\" id=\\"login-role-operator\\" checked=\\"true\\" class=\\"login__option\\"><span class=\\"login__option-content\\"><vuecomponent-stub size=\\"s\\" name=\\"icon-bubble\\" color=\\"default\\" class=\\"login__option-icon\\"></vuecomponent-stub><strong class=\\"login__option-name\\">Operator</strong><span class=\\"login__option-desc paragraph paragraph_color_darkgray\\">Service desk and chat permissions</span></span></baseoption-stub>
      </li>
    </ul>
    <div class=\\"login__form-inputs\\">
      <baseformgroup-stub validator=\\"[object Object]\\" messages=\\"[object Object]\\" label=\\"First name\\" attribute=\\"\\" validatorparams=\\"[object Object]\\" type=\\"text\\" id=\\"sign-up-first-name\\" value=\\"\\" placeholder=\\"Enter your first name\\" class=\\"login__form-group\\"></baseformgroup-stub>
      <baseformgroup-stub validator=\\"[object Object]\\" messages=\\"[object Object]\\" label=\\"Last name\\" attribute=\\"\\" validatorparams=\\"[object Object]\\" type=\\"text\\" id=\\"sign-up-last-name\\" value=\\"\\" placeholder=\\"Enter your last name\\" class=\\"login__form-group\\"></baseformgroup-stub>
      <baseformgroup-stub validator=\\"[object Object]\\" messages=\\"[object Object]\\" label=\\"Email address\\" attribute=\\"\\" validatorparams=\\"[object Object]\\" type=\\"email\\" id=\\"sign-up-email\\" value=\\"\\" placeholder=\\"Enter your email\\" class=\\"login__form-group\\"></baseformgroup-stub>
      <baseformgroup-stub validator=\\"[object Object]\\" messages=\\"[object Object]\\" label=\\"Password\\" attribute=\\"\\" validatorparams=\\"[object Object]\\" type=\\"password\\" id=\\"sign-up-password\\" value=\\"\\" placeholder=\\"Enter your password\\" class=\\"login__form-group\\"></baseformgroup-stub>
    </div>
    <basebutton-stub type=\\"submit\\" icon=\\"\\" theme=\\"default\\" class=\\"login__submit\\">Create Account</basebutton-stub>
    <p class=\\"login__text paragraph\\">Already have an account?<baselink-stub theme=\\"default\\" isbutton=\\"true\\" type=\\"button\\" class=\\"login__changer\\">Sign in</baselink-stub>
    </p>
  </form>
</div>"
`;
